src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1012,6 +1013,20 @@ public static void onEmptyLeftClick(EntityPlayer player, ItemStack stack)
         MinecraftForge.EVENT_BUS.post(new PlayerInteractEvent.LeftClickEmpty(player, stack));
     }
 
+    public static CriticalHitEvent getCriticalHit(EntityPlayer player, Entity target, boolean vanillaCritical, float damageModifier)
+    {
+        CriticalHitEvent hitResult = new CriticalHitEvent(player, target, damageModifier);
+        MinecraftForge.EVENT_BUS.post(hitResult);
+        if (hitResult.getResult() != Event.Result.DENY)
```

Comment:
This `DENY` check is unnecessary. This whole thing can be shortened into

```java
if (hitResult.getResult() == Event.Result.ALLOW || vanillaCrticial)
{
    return hitResult;
} 
return null;
```

Or you could change it to use a ternary operator if you want

```java
return hitResult.getResult() == Event.Result.ALLOW || vanillaCritical ? hitResult : null;
```
```
```